plugins/woocommerce/includes/class-wc-form-handler.php

```diff
@@ -218,9 +216,16 @@ public static function save_address() {
 		 * @since 3.6.0
 		 * @param int    $user_id User ID being saved.
 		 * @param string $address_type Type of address; 'billing' or 'shipping'.
+		 * @param array  $address The address fields. Since 9.8.0.
+		 * @param WC_Customer $customer The customer object being saved. Since 9.8.0.
 		 */
-		do_action( 'woocommerce_customer_save_address', $user_id, $address_type );
+		do_action( 'woocommerce_customer_save_address', $user_id, $address_type, $address, $customer );
 
+		if ( 0 < wc_notice_count( 'error' ) ) {
```

Comment:
This is so we can validate additional fields after the main customer fields have been persisted. This is so we can show the correct fields in the form even if a validation error occurs.